_layouts/base.html

```diff
@@ -4,6 +4,8 @@
   <meta charset="utf-8">
   <meta http-equiv="X-UA-Compatible" content="IE=edge">
 
+  <link rel="canoncial" href="{{ page.url | remove:'index.html' | remove:'.html' | absolute_url }}">
```

Comment:
Yes, I'm aware of the permalink functionality. Pretty links are supported by default in Jekyll, but you can also add a filename and extension to reach the same file. This merely represents a canonicalization so that `/docs/quickstarts/inventory` and `/docs/quickstarts/inventory.html` are represented as the same document.

Jekyll's `link` tag doesn't offer pretty link generation out of the box. (You'd need a custom tag which GitHub Pages doesn't support.) Instead, Jekyll's `link` tag produces a URL with an extension (or the filename and extension in the case of index files) when `permalink` isn't present.

So to answer your question, yes `permalink` front-matter would eliminate the need to do this, but I don't think the trade-off is worth it. You're trading 1 place with this logic for N places specifying permalinks that are largely just duplications of data that is already inferred.

If you feel it'd be cleaner to use permalinks though, I'll do that.